web/src/stream_data.ts

```diff
@@ -826,9 +846,13 @@ export function initialize(params) {
 
     color_data.claim_colors(subscriptions);
 
-    function populate_subscriptions(subs, subscribed, previously_subscribed) {
+    function populate_subscriptions(
+        subs: ApiStreamSubscription[] | NeverSubscribedStream[],
+        subscribed: boolean,
+        previously_subscribed: boolean,
+    ): void {
         for (const sub of subs) {
-            const attrs = {
+            const attrs: StreamAttrs = {
```

Comment:
I found the name `StreamAttrs` a bit confusing, and it is not necessary to define `attrs` prior to calling `create_sub_from_server_data`.

We can inline this type at `create_sub_from_server_data` so there is no need to define a separate type for this.

Reply:
Yeah we can inline this but I thought `StreamAttrs` was a little complex to look at first so I thought it maybe a good idea to define it outside function for better readability?

Reply:
That's reasonable too, but maybe with a better name like `StreamServerData`?

Reply:
I think this being a name like `ApiGenericStreamSubscription`  which is a union of types like `ApiNeverSubscribedStreamSubscription` might be a lot more readable?

Or maybe this doesn't need a type -- shouldn't it be inferred from the `subs` parameter anyway?

Reply:
I like the name `ApiGenericStreamSubscription`, so changed it to this name for now.